[Keycloak CI] - Cookies Tests

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

```
2025-02-26 23:47:25  -- [ERROR] Failures: 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTokenExchangeFederatedTest.testExternalInternalTokenExchangeUsingIssuer
2025-02-26 23:47:25  -- [ERROR]   Run 1: KcOidcBrokerTokenExchangeFederatedTest>KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer:91->KcOidcBrokerTokenExchangeTest.assertExternalToInternalExchange:151 
2025-02-26 23:47:25  -- Expected: <200>
2025-02-26 23:47:25  --      but: was <400>
2025-02-26 23:47:25  -- [ERROR]   Run 2: KcOidcBrokerTokenExchangeFederatedTest>KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer:91->KcOidcBrokerTokenExchangeTest.assertExternalToInternalExchange:151 
2025-02-26 23:47:25  -- Expected: <200>
2025-02-26 23:47:25  --      but: was <400>
2025-02-26 23:47:25  -- [ERROR]   Run 3: KcOidcBrokerTokenExchangeFederatedTest>KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer:91->KcOidcBrokerTokenExchangeTest.assertExternalToInternalExchange:151 
2025-02-26 23:47:25  -- Expected: <200>
2025-02-26 23:47:25  --      but: was <400>
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTokenExchangeV1Test.testExternalInternalTokenExchangeUsingIssuer
2025-02-26 23:47:25  -- [ERROR]   Run 1: KcOidcBrokerTokenExchangeV1Test>KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer:91->KcOidcBrokerTokenExchangeTest.assertExternalToInternalExchange:151 
2025-02-26 23:47:25  -- Expected: <200>
2025-02-26 23:47:25  --      but: was <400>
2025-02-26 23:47:25  -- [ERROR]   Run 2: KcOidcBrokerTokenExchangeV1Test>KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer:91->KcOidcBrokerTokenExchangeTest.assertExternalToInternalExchange:151 
2025-02-26 23:47:25  -- Expected: <200>
2025-02-26 23:47:25  --      but: was <400>
2025-02-26 23:47:25  -- [ERROR]   Run 3: KcOidcBrokerTokenExchangeV1Test>KcOidcBrokerTokenExchangeTest.testExternalInternalTokenExchangeUsingIssuer:91->KcOidcBrokerTokenExchangeTest.assertExternalToInternalExchange:151 
2025-02-26 23:47:25  -- Expected: <200>
2025-02-26 23:47:25  --      but: was <400>
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.organization.broker.OrganizationOIDCBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage
2025-02-26 23:47:25  -- [ERROR]   Run 1: OrganizationOIDCBrokerSelfRegistrationTest>AbstractBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage:748
2025-02-26 23:47:25  -- [ERROR]   Run 2: OrganizationOIDCBrokerSelfRegistrationTest>AbstractBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage:748
2025-02-26 23:47:25  -- [ERROR]   Run 3: OrganizationOIDCBrokerSelfRegistrationTest>AbstractBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage:748
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.organization.broker.OrganizationSAMLBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage
2025-02-26 23:47:25  -- [ERROR]   Run 1: OrganizationSAMLBrokerSelfRegistrationTest>AbstractBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage:748
2025-02-26 23:47:25  -- [ERROR]   Run 2: OrganizationSAMLBrokerSelfRegistrationTest>AbstractBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage:748
2025-02-26 23:47:25  -- [ERROR]   Run 3: OrganizationSAMLBrokerSelfRegistrationTest>AbstractBrokerSelfRegistrationTest.testRememberOrganizationWhenReloadingLoginPage:748
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] Errors: 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTransientSessionsTest.offlineTokenBrowserFlow
2025-02-26 23:47:25  -- [ERROR]   Run 1: KcOidcBrokerTransientSessionsTest.offlineTokenBrowserFlow:604->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [ERROR]   Run 2: KcOidcBrokerTransientSessionsTest.offlineTokenBrowserFlow:604->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [ERROR]   Run 3: KcOidcBrokerTransientSessionsTest.offlineTokenBrowserFlow:604->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTransientSessionsTest.testSingleSignOn
2025-02-26 23:47:25  -- [ERROR]   Run 1: KcOidcBrokerTransientSessionsTest.testSingleSignOn:476->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [ERROR]   Run 2: KcOidcBrokerTransientSessionsTest.testSingleSignOn:476->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [ERROR]   Run 3: KcOidcBrokerTransientSessionsTest.testSingleSignOn:476->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.KcOidcBrokerTransientSessionsTest.testUserInfoEndpoint
2025-02-26 23:47:25  -- [ERROR]   Run 1: KcOidcBrokerTransientSessionsTest.testUserInfoEndpoint:553->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [ERROR]   Run 2: KcOidcBrokerTransientSessionsTest.testUserInfoEndpoint:553->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [ERROR]   Run 3: KcOidcBrokerTransientSessionsTest.testUserInfoEndpoint:553->loginWithBrokerUsingOAuthClient:579 » IllegalArgument Keys to send should be a not null CharSequence
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.OidcClaimToUserSessionNoteMapperTest.claimIsNotPropagatedWhenNameDoesNotMatch
2025-02-26 23:47:25  -- [ERROR]   Run 1: OidcClaimToUserSessionNoteMapperTest.claimIsNotPropagatedWhenNameDoesNotMatch:129->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [ERROR]   Run 2: OidcClaimToUserSessionNoteMapperTest.claimIsNotPropagatedWhenNameDoesNotMatch:129->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [ERROR]   Run 3: OidcClaimToUserSessionNoteMapperTest.claimIsNotPropagatedWhenNameDoesNotMatch:129->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnAllLoginsWhenNameMatchesAndSyncModeIsForce
2025-02-26 23:47:25  -- [ERROR]   Run 1: OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnAllLoginsWhenNameMatchesAndSyncModeIsForce:110->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [ERROR]   Run 2: OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnAllLoginsWhenNameMatchesAndSyncModeIsForce:110->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [ERROR]   Run 3: OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnAllLoginsWhenNameMatchesAndSyncModeIsForce:110->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [INFO] 
2025-02-26 23:47:25  -- [ERROR] org.keycloak.testsuite.broker.OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnFirstLoginOnlyWhenNameMatchesAndSyncModeIsImport
2025-02-26 23:47:25  -- [ERROR]   Run 1: OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnFirstLoginOnlyWhenNameMatchesAndSyncModeIsImport:93->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [ERROR]   Run 2: OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnFirstLoginOnlyWhenNameMatchesAndSyncModeIsImport:93->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
2025-02-26 23:47:25  -- [ERROR]   Run 3: OidcClaimToUserSessionNoteMapperTest.claimIsPropagatedOnFirstLoginOnlyWhenNameMatchesAndSyncModeIsImport:93->login:144 » NullPointer Cannot invoke "java.util.Map.get(Object)" because "this.params" is null
```

https://keycloak-jenkins/view/Keycloak/job/keycloak-continuous-testing/160/execution/node/777/log/

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/UpdateAccountInformationPage.java
+++ b/testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/pages/UpdateAccountInformationPage.java
@@ -97,7 +97,7 @@ public class UpdateAccountInformationPage extends LanguageComboboxAwarePage {
 
     @Override
     public boolean isCurrent() {
-        return PageUtils.getPageTitle(driver).equalsIgnoreCase("update account information");
+        return PageUtils.getPageTitle(driver) != null && PageUtils.getPageTitle(driver).equalsIgnoreCase("update account information");
     }
 
     public String getLabelForField(String fieldId) {

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBaseBrokerTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/AbstractBaseBrokerTest.java
@@ -269,6 +269,12 @@ public abstract class AbstractBaseBrokerTest extends AbstractKeycloakTest {
 
         oauth.getDriver().findElement(By.id("social-" + brokerId)).click();
         oauth.fillLoginForm(username, password);
+
+        if (updateAccountInformationPage.isCurrent()) {
+            log.debug("Updating info on updateAccount page");
+            updateAccountInformationPage.updateAccountInformation(bc.getUserLogin(), bc.getUserEmail(), "Firstname", "Lastname");
+        }
+
         return oauth.parseLoginResponse();
     }
 

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcOidcBrokerTransientSessionsTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/KcOidcBrokerTransientSessionsTest.java
@@ -488,9 +488,7 @@ public final class KcOidcBrokerTransientSessionsTest extends AbstractAdvancedBro
             oauth.clientId(CONSUMER_BROKER_APP_CLIENT_ID);
             oauth.realm(bc.consumerRealmName());
             doLoginSocial(oauth, bc.getIDPAlias(), bc.getUserLogin(), bc.getUserPassword());
-            events.clear();
 
-            updateAccountInformationPage.updateAccountInformation(bc.getUserLogin(), bc.getUserEmail());
             WaitUtils.waitForPageToLoad();
             consentPage.assertCurrent();
             consentPage.confirm();
@@ -498,10 +496,10 @@ public final class KcOidcBrokerTransientSessionsTest extends AbstractAdvancedBro
             EventRepresentation loginEvent;
             do {
                 loginEvent = events.poll();
-            } while (loginEvent != null && ! Objects.equals(EventType.LOGIN.name(), loginEvent.getType()));
+            } while (loginEvent != null && (!Objects.equals(EventType.LOGIN.name(), loginEvent.getType()) ||
+                    !Objects.equals(loginEvent.getClientId(), CONSUMER_BROKER_APP_CLIENT_ID)));
 
             assertThat(loginEvent, notNullValue());
-            assertThat(loginEvent.getClientId(), is(CONSUMER_BROKER_APP_CLIENT_ID));
             assertThat(loginEvent.getUserId(), Matchers.containsString(LightweightUserAdapter.ID_PREFIX));
 
             final String lwUserId = loginEvent.getUserId();
@@ -572,17 +570,16 @@ public final class KcOidcBrokerTransientSessionsTest extends AbstractAdvancedBro
     private EventRepresentation loginWithBrokerUsingOAuthClient(String consumerClientId) {
         oauth.client(consumerClientId, CONSUMER_BROKER_APP_SECRET);
         oauth.realm(bc.consumerRealmName());
+
         doLoginSocial(oauth, bc.getIDPAlias(), bc.getUserLogin(), bc.getUserPassword());
-        events.clear();
-        updateAccountInformationPage.updateAccountInformation(bc.getUserLogin(), bc.getUserEmail(), null, null);
 
         EventRepresentation loginEvent;
         do {
             loginEvent = events.poll();
-        } while (loginEvent != null && ! Objects.equals(EventType.LOGIN.name(), loginEvent.getType()));
+        } while (loginEvent != null && (!Objects.equals(EventType.LOGIN.name(), loginEvent.getType()) ||
+                !Objects.equals(loginEvent.getClientId(), consumerClientId)));
 
         assertThat(loginEvent, notNullValue());
-        assertThat(loginEvent.getClientId(), is(consumerClientId));
         assertThat(loginEvent.getUserId(), Matchers.containsString(LightweightUserAdapter.ID_PREFIX));
 
         return loginEvent;

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/OidcClaimToUserSessionNoteMapperTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/OidcClaimToUserSessionNoteMapperTest.java
@@ -138,7 +138,7 @@ public class OidcClaimToUserSessionNoteMapperTest extends AbstractIdentityProvid
     private AccessToken login() {
         oauth.realm(bc.consumerRealmName())
                 .client("broker-app", consumerClientRep.getSecret())
-                .redirectUri(getAuthServerRoot() + "realms/" + bc.consumerRealmName() + "/account");
+                .redirectUri(getAuthServerRoot() + "realms/" + bc.consumerRealmName() + "/app");
         AuthorizationEndpointResponse authzResponse = doLoginSocial(oauth, bc.getIDPAlias(), bc.getUserLogin(), bc.getUserPassword());
 
         String code = authzResponse.getCode();

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/BackchannelLogoutTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/BackchannelLogoutTest.java
@@ -1,6 +1,5 @@
 package org.keycloak.testsuite.oauth;
 
-import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.CoreMatchers.is;
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.Assert.assertEquals;
@@ -11,7 +10,6 @@ import static org.keycloak.testsuite.admin.ApiUtil.resetUserPassword;
 import static org.keycloak.testsuite.broker.BrokerTestTools.getConsumerRoot;
 import static org.keycloak.testsuite.util.WaitUtils.waitUntilElement;
 
-import org.apache.http.client.methods.CloseableHttpResponse;
 import org.jboss.arquillian.drone.api.annotation.Drone;
 import org.junit.Before;
 import org.junit.Rule;
@@ -38,7 +36,6 @@ import org.keycloak.testsuite.broker.NestedBrokerConfiguration;
 import org.keycloak.testsuite.broker.OidcBackchannelLogoutBrokerConfiguration;
 import org.keycloak.testsuite.util.CredentialBuilder;
 import org.keycloak.testsuite.util.LogoutTokenUtil;
-import org.keycloak.testsuite.util.Matchers;
 import org.keycloak.testsuite.util.WaitUtils;
 import org.keycloak.testsuite.util.oauth.BackchannelLogoutResponse;
 import org.keycloak.testsuite.util.oauth.OAuthClient;
@@ -249,7 +246,7 @@ public class BackchannelLogoutTest extends AbstractNestedBrokerTest {
         OAuthClient oauth2 = oauth.newConfig().driver(driver2);
         oauth2.realm(nbc.consumerRealmName())
                 .clientId(OidcBackchannelLogoutBrokerConfiguration.CONSUMER_CLIENT_ID)
-                .redirectUri(getAuthServerRoot() + "realms/" + nbc.consumerRealmName() + "/account");
+                .redirectUri(getAuthServerRoot() + "realms/" + nbc.consumerRealmName() + "/app");
 
         doLoginSocial(oauth2, nbc.getIDPAlias(), nbc.getUserLogin(), nbc.getUserPassword());
 
@@ -634,6 +631,7 @@ public class BackchannelLogoutTest extends AbstractNestedBrokerTest {
 
         Optional<EventRepresentation> loginEventOptional = eventList.stream()
                 .filter(event -> userId.equals(event.getUserId()))
+                .filter(event -> clientId.equals(event.getClientId()))
                 .filter(event -> event.getType().equals(EventType.LOGIN.name()))
                 .findAny();
 
@@ -825,7 +823,7 @@ public class BackchannelLogoutTest extends AbstractNestedBrokerTest {
         OAuthClient oauth2 = oauth.newConfig().driver(driver2);
         oauth2.realm(nbc.consumerRealmName())
                 .clientId(OidcBackchannelLogoutBrokerConfiguration.CONSUMER_CLIENT_ID)
-                .redirectUri(getAuthServerRoot() + "realms/" + nbc.consumerRealmName() + "/account");
+                .redirectUri(getAuthServerRoot() + "realms/" + nbc.consumerRealmName() + "/app");
         doLoginSocial(oauth2, identityProviderDisplayName, nbc.getUserLogin(), nbc.getUserPassword());
         return oauth2;
     }

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/organization/broker/AbstractBrokerSelfRegistrationTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/organization/broker/AbstractBrokerSelfRegistrationTest.java
@@ -785,9 +785,9 @@ public abstract class AbstractBrokerSelfRegistrationTest extends AbstractOrganiz
         assertFalse(loginPage.isSocialButtonPresent(realmIdp.getAlias()));
 
         driver.navigate().back();
-        // only org idp, back button won't reset the flow and the organization is the same
-        assertTrue(loginPage.isSocialButtonPresent(orgIdp.getAlias()));
-        assertFalse(loginPage.isSocialButtonPresent(realmIdp.getAlias()));
+        // chrome requires refresh, otherwise Sign in button is not active
+        driver.navigate().refresh();
+
         loginPage.loginUsername("test");
         // both realm and org idps because the user does not map to any organization
         assertTrue(loginPage.isSocialButtonPresent(orgIdp.getAlias()));

